---
layout: page
title: Setup
permalink: /setup/
---

This page contains instructions and links for downloading the resources and tools that will be needed for the course.  Step-by-step instructions for most of the software tools can be found on their respective websites, so we have not included that information here.

## Data

The paper by [SÃ¸rlie et al.](http://www.pnas.org/content/100/14/8418.full) referenced throughout this course can be found freely available online from the Proceedings of the National Academy of Sciences.  Be sure to also view the supporting information under the "SI" tab, particularly Supporting Figure 6, Supporting Table 3, and the Supporting Materials and Methods.

As stated in the paper, additional data for the paper are hosted on the [Stanford Genomics Breast Cancer Consortium portal](http://genome-www.stanford.edu/breast_cancer/). The most complete version of Supporting Figure 6 is also available through this portal.  Three files used to create Supporting Figure 6 will be needed:

 * [SupplFigure6.cdt](http://genome-www.stanford.edu/breast_cancer/robustness/data/Suppl_Figure6.cdt) that includes the complete log2 normalized gene expression data used in the hierarchical clustering process.
 * [SupplFigure6.gtr](http://genome-www.stanford.edu/breast_cancer/robustness/data/Suppl_Figure6.gtr) that includes the gene tree generated by the clustering algorithm
 * [SupplFigure6.atr](http://genome-www.stanford.edu/breast_cancer/robustness/data/Suppl_Figure6.atr) that includes the array tree generated by the clustering algorithm

## TreeView

The TreeView program to be used in [Lesson 3]({{ page.root }}/03-figure-treeview/) and [Lesson 5]({{ page.root }}/05-analysis-cluster/) can be found [here](https://bitbucket.org/TreeView3Dev/treeview3). Please download the current release (which was Alpha 3 from July 5, 2016 at the time of writing) and use the JAR file to run the program.  Your computer will need to have Java 7 or higher.

## Cluster

The Cluster program was originally developed by Michael Eisen at Stanford University. A recent implementation ([Cluster3.0](http://bonsai.hgc.jp/~mdehoon/software/cluster/software.htm)) has been made available by Hoon et al., which will be used in [Lesson 5]({{ page.root }}/05-analysis-cluster/).  Please download the appropriate installer for your environment, e.g. the [installer for Windows](http://bonsai.hgc.jp/~mdehoon/software/cluster/clustersetup.exe), and run the .exe file to install Cluster 3.0 under Programs.  You can then run Cluster 3.0 from its installation directory or the start menu.

## R

Several lessons will use the [R language](https://www.r-project.org/) to process data and create visualizations.  Installation packages and instructions for R can be obtained from its [website](https://www.r-project.org/), where versions are available for Microsoft Windows, MacOS, and Linux/UNIX platforms.  Once installed, R can be run in an interactive session where individual commands can be input one at a time.  In addition, R commands can be placed into a script file, and various chunks of the file (up to and including the entire file) can be executed at once.

> ## IDE
> Managing multiple R script files is much easier with an integrated development environment (IDE) such as RStudio, but it is not required for these lessons.
{: .callout}

Once the R environment is up and running, you will need to install several additional packages to be used in [Lesson 4]({{ page.root }}/04-figure-hive-plot/) and [Lesson 6]({{ page.root}}/06-analysis-r/):

 * [HiveR](https://cran.r-project.org/package=HiveR)
 * [dendextend](https://cran.r-project.org/package=dendextend)
 * [dendextendRcpp](https://cran.r-project.org/package=dendextendRcpp)
 * [NMF](https://cran.r-project.org/package=NMF)

These packages can be installed directly from R by running the following commands in an R session:

~~~
install.packages("HiveR")
install.packages("dendextend")
install.packages("dendextendRcpp")
install.packages("NMF")
~~~
{: .source}
